.circleci/config.yml

```diff
@@ -27,8 +35,52 @@ jobs:
               --publish-deployment << parameters.semgrep_deployment_id >> \
               --publish-token $SEMGREP_APP_TOKEN
 
+  pip-publish:
+    docker:
+      - image: cimg/python:3.8
+    environment:
+      TRIGGER_TAG: << pipeline.git.tag >>
+    steps:
+      - run:
+          name: clone main repo, publish tag that triggered the pipeline
+          command: |
```

Comment:
Is there a reason why you don't use the checkout step from circleci?
This is cloning the source code for you via ssh :) 
https://circleci.com/docs/2.0/concepts/